public/video-ui/src/components/VideoAssetAdd/formComponents/VideoAssetUrl.js

```diff
@@ -6,7 +6,7 @@ import React from 'react';
 export default class VideoAssetUrl extends React.Component {
 
   onUpdateAssetUrl = (e) => {
-    const newVersion = this.props.video.activeVersion === undefined ? 0 : this.props.video.activeVersion + 1;
+    const newVersion = this.props.video.activeVersion === undefined ? 0 : Math.max(...this.props.video.assets.map(x => x.version)) + 1;
```

Comment:
The pattern we've been following is the client `PUT`s a well formed asset/video/whatever and the server either accepts or rejects it.

We could move to a model where we send the data required to generate the mentioned asset/video/whatever but that would be a bit of a break from convention.